File: frontend/src/types/target.ts

export interface Target {
  id: number;
  target_id: string;
  former_id: string;
  uniprot_id: string;
  target_name: string;
  gene_name: string;
  target_type: string;
  synonyms: string;
  function: string;
  pdb_structure: string;
  bio_class: string;
  ec_number: string;
  sequence: string;
}

export interface TargetListResponse {
  items: Target[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}

File: backend/app/models/target.py
from sqlalchemy import Column, Integer, String, Text
from app.database import Base


class Target(Base):
    __tablename__ = "targets"

    id = Column(Integer, primary_key=True, index=True)

    target_id = Column(String(50), unique=True, index=True)

    former_id = Column(Text)

    uniprot_id = Column(Text)

    target_name = Column(Text)

    gene_name = Column(Text)

    target_type = Column(Text)

    synonyms = Column(Text)

    function = Column(Text)

    pdb_structure = Column(Text)

    bio_class = Column(Text)

    ec_number = Column(Text)

    sequence = Column(Text)